runners/portability/java/build.gradle

```diff
@@ -32,12 +32,19 @@ configurations {
 
 dependencies {
   compile library.java.vendored_guava_26_0_jre
-  compile library.java.hamcrest_library
   compile project(":runners:java-fn-execution")
   compile project(":runners:java-job-service")
   compile project(path: ":sdks:java:harness", configuration: "shadow")
   compile library.java.vendored_grpc_1_26_0
   compile library.java.slf4j_api
+  compile library.java.joda_time
+  compile "org.hamcrest:hamcrest:2.1"
```

Comment:
Test code should not be present on the main classpath. Follow-up: BEAM-11668 (not a blocker for this PR)